### ISSUE TITLE
checking for POSIX regcomp

### ISSUE THREAD

**Issue by polluks** [gwsw/less#241]
````
configure:6101: gcc -o conftest -g -O2   conftest.c  -lncurses >&5
conftest.c:72:1: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
main() { regex_t r; regmatch_t rm; char *text = "xabcy";
^
conftest.c:73:28: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
if (regcomp(&r, "abc", 0)) exit(1);
                           ^
conftest.c:73:28: note: include the header <stdlib.h> or explicitly provide a declaration for 'exit'
1 warning and 1 error generated.
configure:6101: $? = 1
configure: program exited with status 1
````

**Comment by gwsw** [comment]
Thanks, fixed in d575033f9f8aaf0ab2e5e29152013e218bba29c5.

**Comment by polluks** [comment]
Now works with macOS 12
